src/context/dishes/api-dao.ts

```diff
@@ -0,0 +1,127 @@
+import { err, ok, Result } from 'types/result';
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
+import {
+  Dish,
+  DishesDao,
+  DishGrade,
+  DishSubmissionRequest,
+  DishSubmissionResponse,
+} from './types';
+import { DefaultDish, DefaultDishSubmissionResponse } from './constants';
+
+const dishesApiDao = ({
+  getIdTokenFunc,
+  baseApiUrl,
+}: {
+  getIdTokenFunc: () => Promise<string>;
+  baseApiUrl: string;
+}): DishesDao => {
+  const axiosClient = axios.create({
```

Comment:
Just a note for future, maybe we should instantiate this at the app root and pass it in, since we create one each time.

Also we will need a higher timeout, especially with people submitting multiple images. Lets think about what is suitable.